### PR TITLE
MatList switched to ObjectList, bug fixes

### PR DESCRIPTION
**Pull request by nenad-popovic-91**

### CODE REVIEW

File: plugins/lcvcore/src/qmatop.h [hunk 1]
```diff
@@ -20,7 +21,16 @@ class QMatOp : public QObject{
...
 
+    static void appendItem(lv::QmlObjectList* list, QObject* item);
+    static int itemCount(lv::QmlObjectList* list);
+    static QObject* itemAt(lv::QmlObjectList* list, int idx);
+    static void removeItemAt(lv::QmlObjectList* list, int idx);
+    static void clearItems(lv::QmlObjectList* list);
 public slots:
+
+    void setupMatObjectList(lv::QmlObjectList* list, std::vector<QMat*>* dataPtr = nullptr);
+    void setupMatObjectListFromArray(lv::QmlObjectList* list, const QJSValue& matArray);
```
Comment (dinusv) [inline plugins/lcvcore/src/qmatop.h]: We should return a new list from the array.

```
lv::QmlObjectList* createListFromArray(const QJSValue& array);
```

File: plugins/lcvcore/src/qmatop.h [hunk 2]
```diff
@@ -20,7 +21,16 @@ class QMatOp : public QObject{
 
     QMat* nullMat() const;
 
+    static void appendItem(lv::QmlObjectList* list, QObject* item);
+    static int itemCount(lv::QmlObjectList* list);
+    static QObject* itemAt(lv::QmlObjectList* list, int idx);
+    static void removeItemAt(lv::QmlObjectList* list, int idx);
+    static void clearItems(lv::QmlObjectList* list);
```
Comment (dinusv) [inline plugins/lcvcore/src/qmatop.h]: I would wrap these functions into a separate nested class, to separate MatOp functionality vs functionality on lists.

```cpp
public:

   class ListOperations{
       static void appendItem(lv::QmlObjectList* list, QObject* item);
       // ...
   }

File: plugins/live/src/live_plugin.cpp [hunk 3]
```diff
@@ -66,10 +66,10 @@ void LivePlugin::registerTypes(const char *uri){
     qmlRegisterType<lv::TcpAgent>(         uri, 1, 0, "TcpAgent");
     qmlRegisterType<lv::MapArray>(         uri, 1, 0, "MapArray");
     qmlRegisterType<lv::MapArrayData>(     uri, 1, 0, "MapArrayData");
-    qmlRegisterUncreatableType<lv::QmlObjectList>(
+    /*qmlRegisterUncreatableType<lv::QmlObjectList>(
         uri, 1, 0, "QmlObjectList", "QmlObjectList can only be created internally.");
     qmlRegisterUncreatableType<lv::QmlVariantList>(
-        uri, 1, 0, "QmlVariantList", "QmlVariantList can only be created internally.");
+        uri, 1, 0, "QmlVariantList", "QmlVariantList can only be created internally.");*/
```
Comment (dinusv) [inline plugins/live/src/live_plugin.cpp]: We can remove this.

File: samples/photo/stitcher.qml [hunk 4]
```diff
@@ -15,24 +16,32 @@ Column{
...
-        onEntriesAdded: if ( size() === 3 ) stitcher.input = inputList
+
+    ObjectList {
+        id: inputObjectList
+        function initObjectList(){
+            if ( imageLeft.output !== 0 &&  imageMid.output !== 0 && imageRight.output !== 0) {
+                Cv.MatOp.setupMatObjectListFromArray(inputObjectList, [imageLeft.output, imageMid.output, imageRight.output])
+                stitcher.input = inputObjectList
+            }
+        }
```
Comment (dinusv) [inline samples/photo/stitcher.qml]: Here, we can use:

```
property ObjectList inputObjectList : null
```

And in `initObjectList()`, we simply assign the value:

```
inputObjectList = Cv.MatOp.createListFromArray(...)
```

File: plugins/lcvcore/src/qmatop.h [hunk 5]
```diff
@@ -20,7 +21,16 @@ class QMatOp : public QObject{
...
 public slots:
+
+    void setupMatObjectList(lv::QmlObjectList* list, std::vector<QMat*>* dataPtr = nullptr);
```
[the remark this answers is not in the document]
Comment (nenad-popovic-91) [reply plugins/lcvcore/src/qmatop.h]: Maybe we could avoid it by simply passing an empty array to the other one?

**Pull request merged**